### PR TITLE
Explore: Tag and Values for Influx are filtered by the selected measurement

### LINKED ISSUE #17507: Influx: Adhoc filter keys/values is not filtered by measurement in explore logs field
**What happened**:
Used the adhoc filter in the influx logs query field and got tag keys suggestion from all measurements.

**What you expected to happen**:
To only get suggestion of tag keys from selected measurement.

**How to reproduce it (as minimally and precisely as possible)**:
1. Navigate to explore
2. Select influx as datasource
3. Select logs as mode
4. Select measurement and field
5. Choose tag and noticed the query to influx is made without the selected measurement

### PR DESCRIPTION
**Pull request by hugohaggmark**
**What this PR does / why we need it**:
Before this PR we loaded all Tags and Values without filtering on the selected measurement. This PR will now change that.
 
**Which issue(s) this PR fixes**:
Fixes: #17507

**Special notes for your reviewer**:

### CODE REVIEW

File: public/app/plugins/datasource/influxdb/datasource.ts [hunk 1]
```diff
@@ -182,14 +182,14 @@ export default class InfluxDatasource extends DataSourceApi<InfluxQuery, InfluxO
     return this._seriesQuery(interpolated, options).then(_.curry(this.responseParser.parse)(query));
   }
 
-  getTagKeys(options) {
-    const queryBuilder = new InfluxQueryBuilder({ measurement: '', tags: [] }, this.database);
+  getTagKeys(options: any) {
+    const queryBuilder = new InfluxQueryBuilder({ measurement: options.measurement || '', tags: [] }, this.database);
```
Comment (marefr) [inline public/app/plugins/datasource/influxdb/datasource.ts]: `options` seems to be undefined when selecting key in adhoc filter from dashboard.

**Review by marefr — CHANGES_REQUESTED** [review]
Works as expected in explore, but errors in adhoc filter in dashboards, see comment. Great work :+1: As soon as that is fixed should be able to merge this!

File: public/app/plugins/datasource/influxdb/datasource.ts (hunk 1 above)
Comment (hugohaggmark) [reply public/app/plugins/datasource/influxdb/datasource.ts]: Good catch! I'll fix that.

**Review by marefr — APPROVED** [review]
LGTM

File: public/app/plugins/datasource/influxdb/datasource.ts [hunk 2]
```diff
@@ -182,14 +182,16 @@ export default class InfluxDatasource extends DataSourceApi<InfluxQuery, InfluxO
...
+  getTagKeys(options: any) {
+    options = options || {};
+    const queryBuilder = new InfluxQueryBuilder({ measurement: options.measurement || '', tags: [] }, this.database);
     const query = queryBuilder.buildExploreQuery('TAG_KEYS');
     return this.metricFindQuery(query, options);
   }
 
-  getTagValues(options) {
-    const queryBuilder = new InfluxQueryBuilder({ measurement: '', tags: [] }, this.database);
+  getTagValues(options: any) {
```
Comment (kaydelaney) [inline public/app/plugins/datasource/influxdb/datasource.ts]: Nitpick: Would 
```suggestion
  getTagValues(options = {}) {
```
remove the need for the next line?

**Review by kaydelaney — APPROVED** [review]
LGTM!

File: public/app/plugins/datasource/influxdb/datasource.ts (hunk 2 above)
Comment (hugohaggmark) [reply public/app/plugins/datasource/influxdb/datasource.ts]: Awesome, will fix!

**Pull request merged**